tracer/src/Datadog.Trace/TracerManagerFactory.cs

```diff
@@ -47,6 +49,20 @@ internal TracerManager CreateTracerManager(ImmutableTracerSettings settings, Tra
                 runtimeMetrics: null,
                 logSubmissionManager: previous?.DirectLogSubmission,
                 telemetry: null);
+
+            try
+            {
+                NativeInterop.SetApplicationInfoForAppDomain(RuntimeId.Get(), tracer.DefaultServiceName, tracer.Settings.Environment, tracer.Settings.ServiceVersion);
+            }
+            catch (Exception ex)
+            {
+                // We failed to retrieve the runtime from native this can be because:
+                // - P/Invoke issue (unknown dll, unknown entrypoint...)
+                // - We are running in a partial trust environment
+                Log.Warning(ex, "Failed to set the service name for native.");
+            }
+
+            return tracer;
```

Comment:
There's also `CITracerManager`. I guess we don't care about that case, as we don't (currently?) support profiling in CI?

Reply:
I think this method is also called when using `CITracerManagerFactory`?